network/rcmgr.go

```diff
@@ -0,0 +1,136 @@
+package network
+
+import (
+	"errors"
+
+	"github.com/libp2p/go-libp2p-core/peer"
+	"github.com/libp2p/go-libp2p-core/protocol"
+)
+
+var (
+	ErrResourceLimitExceeded = errors.New("resource limit exceeded")
+	ErrResourceScopeClosed   = errors.New("resource scope closed")
+)
+
+// ResourceManager is the interface to the network resource management subsystem
+type ResourceManager interface {
+	// GetSystem retrieves the system wide resource scope
+	GetSystem() ResourceScope
+	// GetDMZ retrieves the DMZ resource scope
+	GetDMZ() ResourceScope
```

Comment:
I still find it super weird to call this DMZ. Can we call it system scope?

Reply:
Well, the system scope is the global hard limits; the DMZ is for floating resources that have not been associated with the appropriate scope yet (inbound connections, unassociated streams, etc).

But agreed, let's find a better name.

Reply:
how about we call it `FloatingScope`?

Reply:
or `TransientScope`

Reply:
renamed to `GetTransient`.